_posts/2022-06-16-building-an-online-store-with-platform-ui-and-hugo.md

```diff
@@ -0,0 +1,569 @@
+
+```html
+<!-- head.html -->
+<link rel="preconnect" href="https://unpkg.com" crossorigin>
+<link rel="stylesheet" href="https://unpkg.com/@ritterim/platform-ui/dist/platform-ui.min.css" crossorigin>
+```
+
+This will include the main CSS stylings for the Platform UI framework. While we are in the `head.html` file, let's add a title for the site:
+
+```html
+<!-- head.html -->
+<link rel="preconnect" href="https://unpkg.com" crossorigin>
+<link rel="stylesheet" href="https://unpkg.com/@ritterim/platform-ui/dist/platform-ui.min.css" crossorigin>
+
+<title>Hugo Store</title>
+```
+
+Platform UI also includes JavaScript functionality for things like accordions, drawers, and modals. To include this in our site, add the CDN link to `~/hugo-store/themes/store-theme/partials/footer.html`:
+
+```html
+<!-- footer.html -->
+<script src="https://unpkg.com/@ritterim/platform-ui/dist/js/platform-ui.min.js" crossorigin defer></script>
+```
+
+Now we can take advantage of everything Platform UI has to offer!
+
+### Header
+
+We can use Platform UI's [site menu](https://platformui.com/docs/menus/site-menu/) to get this set up. Open up `~/hugo-store/themes/store-theme/partials/header.html` and add in the following:
+
+```html
+<!-- header.html -->
+<header id="header" class="header site-header background--lightblue">
+ <div class="max-container h-100">
+   <a class="site-logo p-2 h-100 flex flex--align-center text--bold text--size-lg" href="/">
+      Hugo Store
+   </a>
+ </div>
+</header>
+````
+
+A quick explanation of what is happening:
+- We create a site header by adding the `site-header` class to the `<header>` element.
+- We can also set the background color using the `background--lightblue` class. This is part of Platform UI's [color](https://platformui.com/docs/colors/colors/) classes.
+- `max-container` on the inner `div` applies a centered, [max-width container](https://platformui.com/docs/layout/max-width/) to the header.
+- On the `site-logo` we are applying flex utility classes to center the link text. We are also adding some [typographical](https://platformui.com/docs/typography/typography/) utilities to enlarge (`text--size-lg`) and bold the text (`text--bold`).
+
+For the purposes of this demo, this is all we need, though this could be expanded further by adding in a navigation!
+
+### Category / List Pages
+
+The category pages will be what we use to display all the categories related products. In Hugo, this is as known as a "list" page, as it lists out all the related content. You can rename this file as something other than `list`, but for our example we will keep it as the default.
+
+We can leverage Platform UI's layout [blocks](https://platformui.com/docs/layout/blocks/) to display the products on the page. Blocks enable a responsive grid of rows and columns to create layouts.
+
+Inside `~/hugo-store/themes/store-theme/layouts/_default/list.html`:
+
+```html
+<!-- list.html -->
+{{ '{{' }} define "main" }}
```

Comment:
You may need to wrap your html blocks like this:
```
html intro
{% raw %}
// markup here
{% endraw %}
html outro
```

Otherwise it might eat the curly braces - I'd run it just to be sure that everything is rendering how you'd expect in the code blocks

Reply:
ah did not realize that was a thing - I [found something](https://joshtronic.com/2020/05/24/how-to-escape-curly-brackets-in-liquid-templates/) that said you can wrap only the opening curly braces inside other curly braces to escape them which seems to work:
<img width="741" alt="Screen Shot 2022-06-16 at 4 00 09 PM" src="https://user-images.githubusercontent.com/5217768/174154145-83c83408-acc8-4aa3-96a2-fad8576bd901.png">